README.md

```diff
@@ -10,53 +10,109 @@ See [docs](https://thesofproject.github.io/latest/index.html)
 ### Prerequisites
 
 * Docker
+* CMake (version >= 3.10)
 
 ### Build Instructions
 
-1. Run `autogen.sh`
+1. Create directory in checked out repo for build files:
+```
+mkdir build && cd build
+```
+
+2. Run configuration for your toolchain:
+
+Baytrail:
+
+```
+cmake -DTOOLCHAIN=xtensa-byt-elf -DROOT_DIR=`pwd`/../xtensa-root/xtensa-byt-elf ..
+```
+
+Cherrytrail:
+
+```
+cmake -DTOOLCHAIN=xtensa-cht-elf -DROOT_DIR=`pwd`/../xtensa-root/xtensa-cht-elf ..
+```
+
+Haswell / Broadwell:
+
+```
+cmake -DTOOLCHAIN=xtensa-hsw-elf -DROOT_DIR=`pwd`/../xtensa-root/xtensa-hsw-elf ..
+```
+
+Apollolake:
+
+```
+cmake -DTOOLCHAIN=xtensa-apl-elf -DROOT_DIR=`pwd`/../xtensa-root/xtensa-apl-elf ..
```

Comment:
Looks like you don't have xtensa-apl-elf-* programs in PATH

Reply:
Need to add the following commands to make this work:
````
pbossart@pbossart-mobl3:/data/pbossart/SOF-public/sof-sdk/sof.git/build$ export PATH=/data/pbossart/SOF-public/sof-sdk/xtensa-root/xtensa-bxt-elf/bin/:$PATH
pbossart@pbossart-mobl3:/data/pbossart/SOF-public/sof-sdk/sof.git/build$ cmake -DTOOLCHAIN=xtensa-bxt-elf -DROOT_DIR=`pwd`/../xtensa-root/xtensa-bxt-elf ..
-- Preparing Xtensa toolchain
-- The C compiler identification is GNU 5.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /data/pbossart/SOF-public/sof-sdk/xtensa-root/xtensa-bxt-elf/bin/xtensa-apl-elf-gcc
-- Generating /data/pbossart/SOF-public/sof-sdk/sof.git/generated/include/version.h
-- Configuring done
-- Generating done
-- Build files have been written to: /data/pbossart/SOF-public/sof-sdk/sof.git
````

Reply:
These steps didn't change with buildsystem, they are in full docs there: https://thesofproject.github.io/latest/getting_started/build-guide/build-from-scratch.html#build-toolchain-from-source

This PR is direct replacements for steps that were previously in README.md, for full build for all GCC toolchains I'd have to just copy-paste what we already have on sof-docs.

Reply:
@plbossart 
BTW `The C compiler identification is GNU 5.2.0` - AFAIK we support GCC 7+ for both host gcc and gcc xtensa toolchains. FW may work with gcc 5 but rimage build will fail for sure because of -Wimplicit-fallthrough flag